uvicorn/lifespan/on.py

```diff
@@ -33,18 +37,21 @@ async def startup(self):
         else:
             self.logger.info("Application startup complete.")
 
-    async def shutdown(self):
+    async def shutdown(self) -> None:
         if self.error_occured:
             return
         self.logger.info("Waiting for application shutdown.")
         await self.receive_queue.put({"type": "lifespan.shutdown"})
         await self.shutdown_event.wait()
         self.logger.info("Application shutdown complete.")
 
-    async def main(self):
+    async def main(self) -> None:
         try:
             app = self.config.loaded_app
-            scope = {"type": "lifespan"}
+            scope: LifespanScope = {
+                "type": "lifespan",
+                "asgi": {"version": self.config.asgi_version, "spec_version": "2.0"},
```

Comment:
Interesting! Should we submit this as a separate pull request, so that this focuses strictly on adding type hints?

Reply:
while I completely agree if I leave the scope as is and just add the type then mypy will fail with
`uvicorn/lifespan/on.py:51: error: Key 'asgi' missing for TypedDict "LifespanScope"`

so I PR the full dict in a separate one then once merged we rebase this one and add types ?

Reply:
ok added #754 !